### ISSUE TITLE
Build Application Form for Gigs

### ISSUE THREAD

**Issue by angeljgomezc** [Developer-DAO/DAO-job-board#26]
Every gig in the website will need a way for devs to apply. 

WHAT ARE GIGS?

They are tiny freelance projects for devs to tackle. This could be anything as simple as centering a div to slightly more difficult things like designing a landing page from scratch or connecting a smart contract to a front-end.

WHY AN APPLICATION FORM? 

Because gigs will be internal between people/companies starting the gigs and devs themselves, we need to make it easy for devs to show what they can offer. 

I suggest a simple form where devs need to write:

- WHY they're ideal for the job (like Gitcoin Bounties pages) - this should be a simple 100-word input form where devs can write an appealing proposal.
- PROOF would be extra inputs where devs show they can actually get the job done (or at least make a good try) with links to repos, live sites, figma files, etc.
- WORK PLAN where the devs explain a step-by-step plan of how they will get the job done. This will be optional, but could make it easier for gig creators to know who has a better of how to get the job done. 

All these inputs should make it easier for those starting the gigs to find their ideal candidate. 

There are many examples to follow about application forms for gigs, I'll recommend these:

[Upwork](https://www.upwork.com/) = the biggest freelance site out there, makes it easy for people to apply on both tiny and large projects by letting them set up different pieces of info before sending the application. We should focus on only a tiny description of the dev and a proof of their competence (a link, for example)

[Gitcoin Bounties](https://gitcoin.co/explorer?network=mainnet&idx_status=open&applicants=ALL&order_by=-web3_created) = Some Gitcoin bounties let hackers apply by sending a tiny message. This is a very straightforward form but gets the job done. 

P.S. We also need an application list. But this will a separate issue we will start later.

**Comment by JoviDeCroock** [comment]
Hey,

I assume we'll need a details-view for the `gig` and allow for an authenticated user to create a `gig_application` there?

I would be up to tackle this if no one else is, we'd probably need an additional field in the database apart from `pitch` (WHY) and `content` (WORK PLAN - we could rename that to plan)  we'll probably need something like skills (PROOF) or do we derive this from the  `user` entity?

**Comment by subject026** [comment]
Hey i'm keen to get stuck in too. 

Is there a figma file or anything similar for this project?

**Comment by angeljgomezc** [comment]
> Hey,
> 
> I assume we'll need a details-view for the `gig` and allow for an authenticated user to create a `gig_application` there?
> 
> I would be up to tackle this if no one else is, we'd probably need an additional field in the database apart from `pitch` (WHY) and `content` (WORK PLAN - we could rename that to plan)  we'll probably need something like skills (PROOF) or do we derive this from the  `user` entity?

Yes! We can come up with the back-end change later. So far I'm thinking of building it the front end first following a similar approach to what I'm talking about above - but you're free to share some ideas :)

Would be great if we had a Figma file of how you intend to do this.

As for the form itself, having a separate component that pops up seems like a great idea. But an input form below the Gig description should also work.

I'm down for the wording changed your proposing. Once again, feel free to come up with your own ideas.

**Comment by angeljgomezc** [comment]
> Hey i'm keen to get stuck in too. 
> 
> Is there a figma file or anything similar for this project?

We still don't have a Figma file apart from the original DeveloperDao.com website files (we're basing the design on that)

The design system is something we'll need to work on soon, so everyone is on the same page about designing and developing. 

Also, we'll probably redesigning almost everything once the MVP is launched (right now we're focusing on the basics)

**Comment by JoviDeCroock** [comment]
Haven't really used Figma yet but I'll do an attempt at creating a mockup for the application form! Thanks for the assigning
